### PR TITLE
Rewrite to using Blocky Furniture

### PR DESCRIPTION
**Pull request by Boy0000**

### CODE REVIEW

File: src/main/kotlin/com.mineinabyss.bonfire/systems/BonfireEffectSystem.kt [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
 import com.mineinabyss.idofront.time.ticks
 import org.bukkit.Particle
+import org.bukkit.entity.ItemDisplay
 import org.bukkit.entity.Player
 
 @AutoScan
-class BonfireEffectSystem : RepeatingSystem(1.ticks) {
+class BonfireEffectSystem : RepeatingSystem(10.ticks) {
     private val TargetScope.player by get<Player>()
     private val TargetScope.effect by get<BonfireEffectArea>()
```
Comment (0ffz) [inline src/main/kotlin/com.mineinabyss.bonfire/systems/BonfireEffectSystem.kt:19]: When is BonfireEffectArea added and how do we untrack when player is out of the area? Or is this just a component thats added to all players with a permission or something?

**Review by 0ffz — CHANGES_REQUESTED** [review]
Really love the rewrite, geary aint perfect but this looks a lot cleaner than using the db! Left some minor comments.

File: src/main/kotlin/com/mineinabyss/bonfire/listeners/PlayerListener.kt [hunk 2]
```diff
@@ -1,227 +1,104 @@
...
-                        val height =
-                            respawnBonfireLocation.distance(getHighestAirBlock(respawnBonfireLocation.block).location)
-                        val entitiesOnRespawn = respawnBonfireLocation.world.getNearbyEntities(
-                            respawnCenterLocation, 0.5, height + 0.5, 0.5
-                        )
-                        entitiesOnRespawn.filterIsInstance<Boat>().forEach {
-                            it.remove()
-                        }
+                    val height = loc.distance(getHighestAirBlock(loc.block).location)
+                    loc.getNearbyEntities(0.5, height + 0.5, 0.5).filterIsInstance<Boat>().forEach(Boat::remove)
```
[the remark this answers is not in the document]
Comment (Boy0000) [reply src/main/kotlin/com/mineinabyss/bonfire/listeners/PlayerListener.kt:58]: some old exploit where you could just fast travel ontop of boats...
might not be needed anymore dependin

File: src/main/kotlin/com.mineinabyss.bonfire/systems/BonfireEffectSystem.kt (hunk 1 above)
Comment (Boy0000) [reply src/main/kotlin/com.mineinabyss.bonfire/systems/BonfireEffectSystem.kt]: it is set when they set their spawn at a bonfire and removed whenever they remove their spawn

File: src/main/kotlin/com/mineinabyss/bonfire/extensions/BonfireHelpers.kt [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            runCatching {
+                val stateItem = gearyItems.createItem(bonfire.states.set) ?: return
+                val metadataPacket = ClientboundSetEntityDataPacket(entityId,
+                    listOf(SynchedEntityData.DataValue(22, EntityDataSerializers.ITEM_STACK, CraftItemStack.asNMSCopy(stateItem)))
+                )
+
+
+                com.mineinabyss.bonfire.bonfire.plugin.launch {
+                    delay(3.ticks)
+                   bonfire.bonfirePlayers.mapNotNull { it.toPlayer() }.forEach {
```
[the remark this answers is not in the document]
Comment (Boy0000) [reply src/main/kotlin/com/mineinabyss/bonfire/extensions/BonfireHelpers.kt]: figured that was more efficient than doing Location.distance checks for every player as that is apparently a taxin calculation

File: src/main/kotlin/com.mineinabyss.bonfire/components/BonfireRemoved.kt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package com.mineinabyss.bonfire.components
+
+class BonfireRemoved
```
[the remark this answers is not in the document]
Comment (Boy0000) [reply src/main/kotlin/com.mineinabyss.bonfire/components/BonfireRemoved.kt]: its meant to be set as offline players can be marked with it to get notified when they rejoin

File: src/main/kotlin/com/mineinabyss/bonfire/listeners/PlayerListener.kt [hunk 5]
```diff
@@ -1,227 +1,104 @@
...
-        isCancelled = true //I think we can cancel this event in any situation where we set/unset respawn.
-        // We don't want to have any regular behavior happen.
-    }
+class PlayerListener : Listener {
 
     @EventHandler(ignoreCancelled = true)
-    fun EntityInsideBlockEvent.standingOnBonfire() {
-        if ((block.state as? Campfire)?.isBonfire == true) isCancelled = true
+    fun PlayerBedEnterEvent.enter() {
+        isCancelled = true
```
[the remark this answers is not in the document]
Comment (Boy0000) [reply src/main/kotlin/com/mineinabyss/bonfire/listeners/PlayerListener.kt:33]: imean i guess if someone for some reason would want beds to work if no bonfire was set but

**Pull request closed without merging**